File: rwh/ch03/GuardExample.scala
def sayHi(n: Int, m: String) = n match {
	case _ if n == 5 			=> "Oooooooooo ! You chose 5"
	case _ if n > 5 && n < 10 	=> "Excellent  ! You chose wisely"
	case _                      => m
}

println(sayHi(5, "hello"))
println(sayHi(9, "world"))
println(sayHi(0, "You lose"))

File: rwh/ch03/GuardExample.hs
-- file: ch03/GuardExample.hs

sayHi n _ 
	| n == 5 			= "Ooooooooo ! You chose 5"
	| n > 5 && n < 10 	= "Excellent ! You chose wisely"
sayHi _ m = m

main = do 
		putStrLn $ sayHi 5 "hello"
		putStrLn $ sayHi 9 "world"
		putStrLn $ sayHi 0 "you loose"